src/chonkie/chunker/semantic.py

```diff
@@ -1,223 +1,184 @@
         self.include_delim = include_delim
         self.sep = "✄"
+        self.min_characters_per_sentence = min_characters_per_sentence
+        self.filter_window = filter_window
+        self.filter_polyorder = filter_polyorder
+        self.filter_tolerance = filter_tolerance
 
-        # Initialize with type annotations
-        self.similarity_threshold: Optional[float]
-        self.similarity_percentile: Optional[int]
-        
-        if isinstance(threshold, float):
-            self.similarity_threshold = threshold
-            self.similarity_percentile = None
-        elif isinstance(threshold, int):
-            self.similarity_percentile = threshold
-            self.similarity_threshold = None
-        else:
-            self.similarity_threshold = None
-            self.similarity_percentile = None
-
-        if isinstance(embedding_model, BaseEmbeddings):
-            self.embedding_model = embedding_model
-        elif isinstance(embedding_model, str):
-            from chonkie.embeddings.auto import AutoEmbeddings
-
-            self.embedding_model = AutoEmbeddings.get_embeddings(
-                embedding_model, **kwargs
-            )
-        else:
-            raise ValueError(f"{embedding_model} is not a valid embedding model")
-
-        # Probably the dependency is not installed
-        if self.embedding_model is None:
-            raise ImportError(
-                f"{embedding_model} is not a valid embedding model",
-                "Please install the `semantic` extra to use this feature",
-            )
-
-        # Keeping the tokenizer the same as the sentence model is important
-        # for the group semantic meaning to be calculated properly
-        super().__init__(self.embedding_model.get_tokenizer_or_token_counter())
-
-        # Remove the multiprocessing flag from the base class
+        # Set the multiprocessing flag to False
         self._use_multiprocessing = False
-
+    
     @classmethod
     def from_recipe(cls, 
-                    name: str = "default", 
-                    lang: Optional[str] = "en", 
-                    path: Optional[str] = None, 
-                    embedding_model: Union[str, BaseEmbeddings] = "minishlab/potion-base-8M",
-                    mode: str = "window",
-                    threshold: Union[str, float, int] = "auto",
-                    chunk_size: int = 2048,
-                    similarity_window: int = 1,
-                    min_sentences: int = 1,
-                    min_chunk_size: int = 2,
-                    min_characters_per_sentence: int = 12,
-                    threshold_step: float = 0.01,
-                    **kwargs: Dict[str, Any]    
-                    ) -> "SemanticChunker":
+                   name: str = "default", 
+                   lang: Optional[str] = "en", 
+                   path: Optional[str] = None,
+                   embedding_model: Union[str, BaseEmbeddings] = "minishlab/potion-base-8M",
```

Comment:
we are switching to the 32M everywhere in our codebase, could be a good opportunity to make the switch rn
```suggestion
                   embedding_model: Union[str, BaseEmbeddings] = "minishlab/potion-base-32M",
```